db/seeds.rb

```diff
@@ -4,4 +4,189 @@
 # Examples:
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+
+
+require 'rubygems'
+require 'json/pure'
+
+def getAllCourses
+	contents = File.open("db/s18.csv", "r"){ 
+		|file| file.read 
+	}
+
+	all_courses = JSON.parse(contents)
+	return all_courses
+end
+
+def createSemesters
+	Semester.create(season: 'Spring', year: 2018)
+end
+
+def createSections course, sections
+
+	days = {
+		0 => 'U',
+		1 => 'M',
+		2 => 'T',
+		3 => 'W',
+		4 => 'R',
+		5 => 'F',
+		6 => 'S'
+	}
+
+	sections.each do |sec|
+
+		section = Section.new
+		section.course = course
+		section.name = sec["name"]
+		section.location = sec["times"][0]["location"]
+		section.building = sec["times"][0]["building"]
+		section.room = sec["times"][0]["room"]
+		section.start_time = sec["times"][0]["begin"] ? sec["times"][0]["begin"].to_time : nil
+		section.end_time = sec["times"][0]["end"] ? sec["times"][0]["end"].to_time : nil
+		section.days = sec["times"][0]["days"] ? sec["times"][0]["days"].inject(" ") { |sum, day| sum + days[day] } : nil
+		section.save!
+
+		instructors = sec["instructors"]
+		instructors.each do |i|
+			names = i.split(",")
+			existing_instructor = Instructor.where(fname: names[1], lname: names[0])
+			if existing_instructor.empty?
+				instructor = Instructor.create(fname: names[1], lname: names[0])
+			else
+				instructor = existing_instructor.first
+			end 
+
+			SectionInstructor.create(section: section, instructor: instructor)
+
+		end
+
+	end
+
+end
+
+def createCourses
+	all_courses = getAllCourses()
+
+	# create 2988 Course objects
+	all_courses["courses"].each do |course_code, data|
+		
+		course = Course.new
+	    course.code = course_code
+	    course.name = data["name"]
+	    course.department = data["department"]
+	    course.units = data["units"]
+	    course.description = data["desc"]
+	    course.mini = 0
+	    course.save!
+
+	    createSections(course, data["sections"])
+	end
+end
+
+def createCourseSemesters
+	@sem = Semester.first
+
+	Course.all.each do |course|
+		CourseSemester.create(semester: @sem, course: course)
+	end
+
+end
+
+def createMajors
+	Major.create(name: 'Information Systems')
+end
+
+def createRequirements
+	requirements = [
+		["Mathematics and Statistics 1", "C", 28],
```

Comment:
Is 28 the number of units you need for Math/Stats? Wierd number, so just checking

Reply:
so the number of units is a minimum unit count for that requirement so in this case the sequence mandates three courses of units 9 + 10 + 9 therefore 28